### PR TITLE
Fix 3 simple bugs

### PR DESCRIPTION
**Pull request by git-jiby-me**
- The regex for matching request, was not considering arrays in the
request JSON
- For request with a request body, canned was checking content type to
exactly match `application/json`, which is not good as browsers may
sent charset as well with the content type.
- For matching request and filters with more accuracy, we were
converting the values of all keys in request to string before
comparing, but this was being done wrong as it was creating string of
Objects and arrays as well, which it shouldn’t

### CODE REVIEW

**Comment by git-jiby-me** [comment]
@sideshowcoder I forgot to create a new feature branch for this PR, but i guess that is ok, please review and let me know if you need any changes.

**Comment by git-jiby-me** [comment]
@sideshowcoder I am also working in something similar to https://github.com/sideshowcoder/canned/issues/56, but my requirement being more of like having a filter like 
```
//! proxyRequest: http:domain.com/path/endpoint?queryKey=queryValue
```
which will basically proxy matching request to the url and then provide the response from the url back.

Would this be something interesting for you?

File: canned.js [hunk 1]
```diff
@@ -87,6 +87,16 @@ function getContentType(mimetype){
   return Response.content_types[mimetype]
 }
 
+function traverseAndSanitise(object) {
```
Comment (sideshowcoder) [inline canned.js]: Sorry just a type `traverseAndSanitize`

Comment (git-jiby-me) [inline canned.js]: @sideshowcoder oops, fixing right away

Comment (sideshowcoder) [inline canned.js]: Maybe call it `stringifyValues` because that is what it does I think.

File: canned.js [hunk 2]
```diff
@@ -104,10 +114,7 @@ Canned.prototype.parseMetaData = function(response) {
 
         // Force all requests values to be a string.
         // Otherwise comparison in getSelectedResponse doesn't works
-        _.each(metaData.request, function(value, key){
-          metaData.request[key] = String(value)
-        })
-
+        traverseAndSanitise(metaData.request);
```
Comment (sideshowcoder) [inline canned.js]: Calling it `stringifyValues` makes this comment reduantant :) so we can remove it.

Comment (git-jiby-me) [inline canned.js]: @sideshowcoder okey dokey :+1: , 1 min

File: canned.js [hunk 3]
```diff
@@ -355,7 +364,8 @@ Canned.prototype.responseFilter = function (req, res) {
     })
     req.on('end', function () {
       var responderBody = querystring.parse(body);
-      if (req.headers && req.headers['content-type'] === 'application/json') {
+      if (req.headers && req.headers['content-type'] && 
+                    req.headers['content-type'].indexOf('application/json') !== -1) {
```
Comment (sideshowcoder) [inline canned.js]: Can you move that condition into it's own function aka `isContentTypeJson` this is a little large now

File: canned.js (hunk 1 above)
Comment (git-jiby-me) [inline canned.js]: @sideshowcoder done

File: canned.js (hunk 2 above)
Comment (git-jiby-me) [inline canned.js]: @sideshowcoder done

File: canned.js (hunk 3 above)
Comment (git-jiby-me) [inline canned.js]: @sideshowcoder sure

**Comment by sideshowcoder** [comment]
Yes such a change feature as proxy would be quite valuable! thank you :)

I will try to get out a new version to npm today :)

**Comment by sideshowcoder** [comment]
I added you as a collaborator on canned as you are already adding a bunch of features and fixes :)

File: canned.js (hunk 3 above)
Comment (git-jiby-me) [inline canned.js]: @sideshowcoder done

**Comment by git-jiby-me** [comment]
@sideshowcoder cool, thanks, thats nice :+1:

**Comment by sideshowcoder** [comment]
out on npm.

**Comment by git-jiby-me** [comment]
@sideshowcoder thanks for the quick publish, much appreciated :+1:

**Pull request merged**